### ISSUE TITLE
Documentation for API?

### ISSUE THREAD

**Issue by perfectom** [NginxProxyManager/nginx-proxy-manager#341]
Hello, I see there is some kind of API at "/api" , please is there any documentation for it? I would like to incorporate nginx-proxy-manager into a development tool with automated hosts created. Thanks!

**Comment by FantaBlueMystery** [comment]
Yes and please add to api "get certificate", we use the 'lets encrypt' certificate for email server.
If we can cronjob the certificate with Curl + Shell for email servers, that would be very helpful.

**Comment by chaptergy** [comment]
You can find a swagger [schema in this repo](https://github.com/jc21/nginx-proxy-manager/blob/develop/backend/doc/api.swagger.json) which you can then view e.g. with [editor.swagger.io](https://editor.swagger.io/). 

Or you can clone the repo, and run `docker-compose -f docker/docker-compose.dev.yml up`. This will also start a docker container which contains a server for the swagger files, which is accessible at `localhost:3001`. 

But I think it might be slightly out of date, but maybe sufficient for your purposes.

**Comment by setswei** [comment]
I would also like to see this re-opened. i am working on ansible playbooks to maintain state and execute particular automation workflows that i would like NPM to be a part of.

**Comment by chaptergy** [comment]
Sure, I could reopen it though I doubt this ever gets resolved. As https://github.com/NginxProxyManager/nginx-proxy-manager/discussions/1202 describes, the focus is currently mainly on v3, which has a new approach (or has one at all lol) to the documentation and testing of the API, so it should always be well documented. Creating a documentation for v2 would mean a huge amount of work for a product which will be obsolete hopefully soonish. For now you'll have to make do with what I suggested above.

**Comment by setswei** [comment]
Apologizes @chaptergy, i was not aware version 3 was in the works. Are there plans to allow API access in the new version and if so i am happy to wait and not spend time on version 2 ?

**Comment by kennylajara** [comment]
@chaptergy  Is [this](https://github.com/NginxProxyManager/nginx-proxy-manager/projects/1) the project for Version 3 been updated? Is it updated? I see the last update is from 2019

**Comment by laopaoer-wallet** [comment]
Looking forward to this feature, because it is necessary to integrate other systems

**Comment by webysther** [comment]
For who maybe only want to start with, save the [api.swagger.json](https://github.com/NginxProxyManager/nginx-proxy-manager/blob/develop/backend/doc/api.swagger.json) and import in Postman as a collection, for who is hurry:

![image](https://github.com/NginxProxyManager/nginx-proxy-manager/assets/750007/1a5ad40b-732b-4cef-bc13-6e7e0a2b2005)

[Nginx Proxy Manager API.postman_collection.json.zip](https://github.com/NginxProxyManager/nginx-proxy-manager/files/11583508/Nginx.Proxy.Manager.API.postman_collection.json.zip)

**Comment by ysenn** [comment]
I am looking forward to this function. Is this function available now?

**Comment by ysenn** [comment]
looking forward to this feature. Is this feature available now?

**Comment by aamir-aala** [comment]
Just released https://github.com/aalasolutions/nginx-proxy-manager-sdk let me know if there is any issue or feedback

**Comment by chrisn-au** [comment]
> Just released https://github.com/aalasolutions/nginx-proxy-manager-sdk let me know if there is any issue or feedback

Nice work @YoyoSailer.  Is there any desire to get this included in the base package - it would be great to have it included. In an agentic world configuration will increasingly be done by agents. It would be awesome to have this as part of the core package. I ma happy to build a PR if there is interest (on the shoulders of giants)

**Comment by YoyoSailer** [comment]
> > Just released https://github.com/aalasolutions/nginx-proxy-manager-sdk let me know if there is any issue or feedback
> 
> Nice work [@YoyoSailer](https://github.com/YoyoSailer). Is there any desire to get this included in the base package - it would be great to have it included. In an agentic world configuration will increasingly be done by agents. It would be awesome to have this as part of the core package. I ma happy to build a PR if there is interest (on the shoulders of giants)

I'm more then happy if this goes in, in core. but on the other hand I am not sure if that is the right way. Only developers will be needing that and having this as  a separate package is suitable.  

In short `nginx-proxy-manager` is manager it self
`nginx-proxy-manager-sdk` is how you manage your manager via NODE / JS 

And hmm we can make some postman collection or come up with any other solution, but it depends. I build the sdk as i was in need for this.
